packages/babel-traverse/src/path/introspection.js

```diff
@@ -439,3 +439,29 @@ export function isConstantExpression() {
 
   return false;
 }
+
+export function isInStrictMode() {
+  const start = this.isProgram() ? this : this.parentPath;
+
+  const strictParent = start.find(path => {
+    if (path.isProgram({ sourceType: "module" })) return true;
+
+    if (path.isClass()) return true;
+
+    if (!path.isProgram() && !path.isFunction()) return false;
+
+    let { node } = path;
+    if (path.isFunction()) node = node.body;
```

Comment:
Would it be worth handling cases where the `path` is the function name, or a function parameter? Like technically it could be

```
function NODE(){"use strict";}
// OR
function fn(NODE){"use strict";}
```
too. It isn't valid to having function initializers or patterns at least, so we wouldn't have to check too many cases.

Reply:
Already handled.

> `function NODE(){"use strict";}`

https://github.com/babel/babel/pull/7712/files#diff-e0bc96e619aa8318bff982e8def5245fR82

> `function fn(NODE){"use strict";}`

https://github.com/babel/babel/pull/7712/files#diff-e0bc96e619aa8318bff982e8def5245fR92

Reply:
Note: this is directly related to https://github.com/rwaldron/tc39-notes/blob/master/es9/2018-03/mar-22.md#8ic-expand-text-included-in-function-code which we just talked about in last month's meeting. Consensus reached was the `id` and `params` are strict if the function body is strict.

Reply:
Oh of course, that's what I get for reviewing while sleepy :P